java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -181,4 +197,53 @@ public CompletableFuture<String> set(
         String[] arguments = ArrayUtils.addAll(new String[] {key, value}, options.toArgs());
         return commandManager.submitNewCommand(SetString, arguments, this::handleStringOrNullResponse);
     }
+
+    @Override
+    public CompletableFuture<Long> lpush(@NonNull String key, @NonNull String[] elements) {
+        String[] arguments = ArrayUtils.addFirst(elements, key);
+        return commandManager.submitNewCommand(LPush, arguments, this::handleLongResponse);
+    }
+
+    @Override
+    public CompletableFuture<String> lpop(@NonNull String key) {
+        return commandManager.submitNewCommand(
+                LPop, new String[] {key}, this::handleStringOrNullResponse);
+    }
+
+    @Override
+    public CompletableFuture<String[]> lpopCount(@NonNull String key, long count) {
+        return commandManager
+                .submitNewCommand(
+                        LPop, new String[] {key, Long.toString(count)}, this::handleArrayOrNullResponse)
+                .thenApply(objectArray -> objectArrayToTypedArray(objectArray, String.class));
```

Comment:
Try to combine functions to avoid extra async runtime activities, which reduces performance
```suggestion
                        LPop, new String[] {key, Long.toString(count)}, response -> objectArrayToTypedArray(handleArrayOrNullResponse(response), String.class));
```